hogben/utils.py

```diff
@@ -130,31 +134,40 @@ def __corner(self, results):
         axes = np.reshape(np.array(fig.get_axes()), (self.ndim, self.ndim))
         for i in range(1, self.ndim):
             for j in range(self.ndim):
-                if i == self.ndim-1:
-                    axes[i,j].set_xlabel(self.params[j].name)
+                if i == self.ndim - 1:
+                    axes[i, j].set_xlabel(self.params[j].name)
                 if j == 0:
-                    axes[i,j].set_ylabel(self.params[i].name)
+                    axes[i, j].set_ylabel(self.params[i].name)
 
-        axes[self.ndim-1, self.ndim-1].set_xlabel(self.params[-1].name)
+        axes[self.ndim - 1, self.ndim - 1].set_xlabel(self.params[-1].name)
         return fig
 
-def fisher(qs, xi, counts, models, step=0.005):
+
+def fisher(qs: list[list],
+           xi: list[Union['refnx.analysis.Parameter',
+                          'bumps.parameter.Parameter']],
+           counts: list[int],
+           models: list[Union['refnx.reflect.ReflectModel',
+                              'refl1d.experiment.Experiment']],
+           step: float = 0.005,
+           importance: Optional[list[float]] = None) -> np.ndarray:
     """Calculates the Fisher information matrix for multiple `models`
        containing parameters `xi`.
```

Comment:
Can you make the docstring a little more explicit about what we are doing here - e.g. that we transform the parameters into a range [0,1] defined by their bounds, and then calculate the fisher information of this, before allowing scaling by relative importance of each parameter (so we make it clear that the importance scales the FIM, not the Jacobian, e.g. the variance, not the S.D.)

Reply:
Also make it a little clearer what the multiple models might be